core/lib/zksync_core/src/api_server/web3/namespaces/zks.rs

```diff
@@ -334,17 +334,12 @@ impl ZksNamespace {
 
         let merkle_tree_leaves = all_l1_logs_in_batch.iter().map(L2ToL1Log::to_bytes);
 
-        let min_tree_size = if batch
+        let protocol_version = batch
             .protocol_version
-            .map(|v| v.is_pre_boojum())
-            .unwrap_or(true)
-        {
-            Some(L2ToL1Log::PRE_BOOJUM_MIN_L2_L1_LOGS_TREE_SIZE)
-        } else {
-            Some(L2ToL1Log::MIN_L2_L1_LOGS_TREE_SIZE)
-        };
+            .unwrap_or_else(ProtocolVersionId::last_potentially_undefined);
+        let min_tree_size = min_l2_to_l1_logs_tree_size(protocol_version);
```

Comment:
or alternatively - we could allow min_l2_to_l1_logs_tree_size to accept Option<ProtocolVersionId> instead.

Reply:
ack.